File: backend/internal/models/deal.go
package models

import "time"

type Deal struct {
	ID            int64     `json:"id"`
	ExternalID    string    `json:"external_id"`
	Title         string    `json:"title"`
	Description   string    `json:"description"`
	Price         string    `json:"price"`
	OriginalPrice string    `json:"original_price"`
	ImageURL      string    `json:"image_url"`
	DealURL       string    `json:"deal_url"`
	Merchant      string    `json:"merchant"`
	Temperature   int       `json:"temperature"`
	Category      string    `json:"category"`
	ScrapedAt     time.Time `json:"scraped_at"`
	UpdatedAt     time.Time `json:"updated_at"`
	IsExpired     bool      `json:"is_expired"`
	ClickCount    int64     `json:"click_count"`
}

File: frontend/src/types/deal.ts
export interface Deal {
  id: number
  external_id: string
  title: string
  description: string
  price: string
  original_price: string
  image_url: string
  deal_url: string
  merchant: string
  temperature: number
  category: string
  scraped_at: string
  updated_at: string
  is_expired: boolean
  click_count: number
}

export interface DealsResponse {
  data: Deal[]
  total: number
  limit: number
  offset: number
  has_more: boolean
}

export interface ClickResponse {
  redirect_url: string
}
